idea-plugin/src/main/kotlin/arrow/meta/ide/phases/resolve/FragmentProviderComponent.kt

```diff
@@ -32,15 +33,19 @@ class FragmentProviderComponent(val project: Project) : ProjectComponent, AsyncF
     } ?: LOG.error("Could not get MetaSyntheticPackageFragmentProvider instance, while opening the project")
   }
 
-  override fun prepareChange(events: MutableList<out VFileEvent>): FragmentProviderComponent? {
-    // we only care about changes to .class files inside of build folders
-    val buildFolders = project.buildFolders().toHashSet()
-    val hasBuildFolderEvents = events.any { e ->
-      val file = e.file
-      e.isValid && file != null && file.extension == "class" && VfsUtilCore.isUnder(file, buildFolders)
+  /**
+   *  we only care about changes to .class files inside of build folders
+   */
+  override fun prepareChange(events: MutableList<out VFileEvent>): FragmentProviderComponent? =
+    this.takeIf {
+      project.run {
+        events.any { e: VFileEvent ->
+          e.file?.let { file: VirtualFile ->
+            e.isValid && file.extension == "class" && VfsUtilCore.isUnder(file, buildFolders().toHashSet())
```

Comment:
Could buildFolder() be cached? In a worst-case scenario buildFolders() would be called `n` times for `n` events here . `prepareChange()` is called for bulk processing and buildFolders() isn't using cached data itself.

Reply:
makes sense :) 
Let me change that